harness/determined/estimator/_reducer.py

```diff
@@ -0,0 +1,161 @@
+import abc
+from typing import Any, Callable, List
+
+import numpy as np
+import tensorflow as tf
+
+from determined import estimator
+
+
+class MetricReducer:
```

Comment:
Question: do we want to show this in the docs as well?

Reply:
oops, definitely yes.  Added.